tools/importers/SynchronizeTypographyTokens.ts

```diff
@@ -0,0 +1,189 @@
+	jsonTypography: Object,
+	isTheme: boolean,
+) => {
+	let resultCss: Array<string> = [];
+	let resultScss: Array<string> = [];
+	let resultJsonCss: Dict<Array<IResultJsonObject>> = {};
+
+	resultCss.push(modifiers.cssFileFirstLine(isTheme, name));
+
+	for (let key in jsonTypography) {
+		if (!importerVariables.excludedKeys.includes(key)) {
+			let value: any = jsonTypography[key];
+
+			for (let variableKey in value) {
+				let propertyName: string =
+					importerVariables.typographyPrefix +
+					importerVariables.jsonAttributeToCssProperty[key] +
+					'-' +
+					variableKey;
+				let propertyValue: string | undefined;
+
+				switch (key) {
+					case importerVariables.tokensTypographyInputAttributes.fontSize:
+					case importerVariables.tokensTypographyInputAttributes.lineHeight:
+						const valueBase: number = parseInt(value.base.value.replace(/[^0-9]/g, ''));
+						const valueSplit: Array<number> = value[variableKey].value.split('*');
+						const valueMultiplier: number = valueSplit[1];
+						propertyValue = valueMultiplier
+							? valueBase * valueMultiplier + 'px'
+							: valueBase + 'px';
+						break;
+					case importerVariables.tokensTypographyInputAttributes.letterSpacing:
+						propertyValue =
+							value[variableKey].value.replace(/[^0-9/.]/g, '') * 0.01 + '';
+						if (value[variableKey].value != 0) {
+							propertyValue += 'em';
+						}
+						break;
+					case importerVariables.tokensTypographyInputAttributes.textCase:
+					case importerVariables.tokensTypographyInputAttributes.textDecoration:
+						propertyValue = value[variableKey].value;
+						break;
+				}
+				if (propertyValue !== undefined) {
+					resultCss.push('--' + propertyName + ': ' + propertyValue + ';');
+					resultScss.push('$' + propertyName + ': var(--' + propertyName + ');');
+
+					const resultJsonObject: ITokenJsonObject = {
+						id: binValues.destinationJson + '_' + propertyName,
+						label: propertyName,
+						value: propertyValue,
+					};
+
+					if (resultJsonCss[key] === undefined) {
+						resultJsonCss[key] = [];
+					}
+					resultJsonCss[key].push(resultJsonObject);
+				}
+			}
+		}
+	}
+
+	importerVariables.resultCssAdditionalLines.forEach((item) => {
+		resultCss.push(item);
+	});
+	importerVariables.resultScssAdditionalLines.forEach((item) => {
+		resultScss.push(item);
+	});
+
+	fileWriter.arrayToFile(
+		tokensFilesConfig.destinationPath + binValues.destinationVariablesCss,
+		resultCss,
+	);
+	fileWriter.arrayToFile(
+		tokensFilesConfig.destinationPath + binValues.destinationVariables,
+		resultScss,
+	);
+	fileWriter.jsonToFile(
+		tokensFilesConfig.destinationPath + binValues.destinationVariablesCssJson,
+		resultJsonCss,
+	);
+};
+
+const ImportTypographyTokensRaw = (binValues: ConfigFileObject, jsonTypography: any) => {
+	let resultScss: Array<ITypographyToken> = [];
+
+	for (let key in jsonTypography[importerVariables.tokensKey]) {
+		let token: ITypographyToken = fileRead.recursiveTokenReader(
```

Comment:
why is it called `fileRead(er)`? It isn't reading a file.

Reply:
Because I'm reading the file content there. Maybe `fileCrawler` @kvas-damian ?

Reply:
we're not touching file. We operate on a tree of variables. I wouldn't call it `fileAnything`

Reply:
Renamed filed to `tokensReader.ts`